sdk/search/Azure.Search.Documents/src/Indexes/FieldBuilder.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Core;
+using Azure.Search.Documents.Indexes.Models;
+
+namespace Azure.Search.Documents.Indexes
+{
+    /// <summary>
+    /// Builds a list of <see cref="SearchField"/> given a model type for use in a <see cref="SearchIndex"/>.
+    /// <seealso cref="SimpleFieldAttribute"/>
+    /// <seealso cref="SearchableFieldAttribute"/>
+    /// </summary>
+    public static class FieldBuilder
+    {
+        private static readonly DefaultNamingPolicy s_defaultNamingPolicy = new DefaultNamingPolicy();
+
+        /// <summary>
+        /// Builds a list of <see cref="SearchField"/> given a model type <typeparamref name="T"/> for use in a <see cref="SearchIndex"/>.
+        /// By default, all public properties and fields of that model type will generate a <see cref="SearchField"/>;
+        /// however, you can ignore them using <see cref="JsonIgnoreAttribute"/> or <see cref="NonSerializedAttribute"/>.
+        /// You can further customize the generated <see cref="SearchField"/> using the
+        /// <see cref="SimpleFieldAttribute"/> and <see cref="SearchableFieldAttribute"/> attributes.
+        /// </summary>
+        /// <typeparam name="T">The type of model from which to build fields.</typeparam>
+        /// <returns>A list of <see cref="SearchField"/> from model type <typeparamref name="T"/>.</returns>
+        public static IList<SearchField> Build<T>() => Build(typeof(T), s_defaultNamingPolicy);
+
+        /// <summary>
+        /// Builds a list of <see cref="SearchField"/> given a model type <typeparamref name="T"/> for use in a <see cref="SearchIndex"/>.
+        /// By default, all public properties and fields of that model type will generate a <see cref="SearchField"/>;
+        /// however, you can ignore them using <see cref="JsonIgnoreAttribute"/> or <see cref="NonSerializedAttribute"/>.
+        /// You can further customize the generated <see cref="SearchField"/> using the
+        /// <see cref="SimpleFieldAttribute"/> and <see cref="SearchableFieldAttribute"/> attributes.
+        /// </summary>
+        /// <typeparam name="T">The type of model from which to build fields.</typeparam>
+        /// <param name="namingPolicy">
+        /// The <see cref="JsonNamingPolicy"/> to use for converting property names if not already attributed with <see cref="JsonPropertyNameAttribute"/>.
+        /// </param>
+        /// <returns>A list of <see cref="SearchField"/> from model type <typeparamref name="T"/>.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="namingPolicy"/> is null.</exception>
+        public static IList<SearchField> Build<T>(JsonNamingPolicy namingPolicy) => Build(typeof(T), namingPolicy);
```

Comment:
I think there's going to be some iteration on the JsonNamingPolicy vs ObjectSerializer parameter type.  I would lean toward not including these now to avoid having to potentially remove them next week if we're not sure about the direction by Code Complete.

Reply:
Okay. I'll just remove the overloads entirely for now. Can always add them later if we still haven't decided by GA>